ci/scripts/check_bazel_target_names_test.py

```diff
@@ -0,0 +1,101 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import unittest
+from unittest.mock import Mock
+
+from check_bazel_target_names import BazelQueryRunner
+
+
+class TestFindTargetsWithBannedChars(unittest.TestCase):
+
+    def test_no_test_suites(self):
+        backend = Mock()
+        backend.return_value = []
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [])
+
+    def test_one_target_empty_string(self):
+        """Bazel does not allow targets to have empty names.
+
+        This test simply documents how we respond to the impossible scenario.
+        """
+        backend = Mock()
+        backend.return_value = [""]
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [])
+
+    def test_only_good_chars(self):
+        backend = Mock()
+        backend.return_value = ["//foo:bar", "//bar_baz:foo"]
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [])
+
+    def test_only_bad_chars(self):
+        backend = Mock()
+        backend.return_value = ["!@#$", "^&*()", "\x01"]
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [
+            ("!@#$", set("!@#$")),
+            ("^&*()", set("^&*()")),
+            ("\x01", set("\x01")),
+        ])
+
+    def test_mixed(self):
+        backend = Mock()
+        backend.return_value = [
+            '!@#$', '\x01', '//foo:bar', '^&*()', '//bar_baz:foo'
+        ]
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [
+            ("!@#$", set("!@#$")),
+            ("\x01", set("\x01")),
+            ("^&*()", set("^&*()")),
+        ])
+
+
+class TestFindEmptyTestSuites(unittest.TestCase):
```

Comment:
This seems like it should be beyond the scope of "check_bazel_targe_names"

Reply:
Yeah, I admit it's a stretch. Maybe I'll split it out into two scripts.